Composer/packages/client/src/pages/language-generation/code-editor.tsx

```diff
@@ -8,17 +8,19 @@ import get from 'lodash/get';
 import debounce from 'lodash/debounce';
 import isEmpty from 'lodash/isEmpty';
 import { CodeRange, LgFile } from '@bfc/shared';
+import { editor } from '@bfcomposer/monaco-editor/esm/vs/editor/editor.api';
 
 import * as lgUtil from '../../utils/lgUtil';
 
 interface CodeEditorProps {
   file: LgFile;
   onChange: (value: string) => void;
-  codeRange?: Partial<CodeRange> | null;
+  codeRange?: Partial<CodeRange> | undefined;
```

Comment:
I think you don't need `| undefined`, because `?` implies that.

```suggestion
  codeRange?: Partial<CodeRange>;
```